shtdlib.sh

```diff
@@ -796,20 +796,39 @@ function init_nss_wrapper {
     export BGID="${GUID#*:}"
     passwd_string="${TMP_USER}:x:${BUID}:${BGID}:Bob the builder:${TMP_HOME_PATH}:/bin/false"
     group_string="${TMP_GROUP}:x:${BUID}:"
-    passwd_pattern=".*:x:${BUID}:.*:.*:.*:.*"
-    group_pattern=".*:x:${BGID}:.*"
+    passwd_name_pattern='^'"${TMP_USER}"':x:.*:.*:.*:.*:.*$'
+    passwd_pattern='^.*:x:'"${BUID}"':.*:.*:.*:.*$'
+    group_name_pattern="^${TMP_GROUP}"':x:.*:.*$'
+    group_pattern='^.*:x:'"${BGID}"':.*$'
+
+    if grep -qE "${passwd_name_pattern}" "${tmp_passwd_file}"; then
+      sed -i "s|${passwd_pattern}||g" "${tmp_passwd_file}"
```

Comment:
Suggesting perhaps merging this block with the code that does the group editing (it's the same pattern, just a variable difference).